File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic/BoundXmlEmbeddedExpression.cs
using System.Diagnostics;
using Microsoft.CodeAnalysis.VisualBasic.Symbols;

namespace Microsoft.CodeAnalysis.VisualBasic
{
	internal sealed class BoundXmlEmbeddedExpression : BoundExpression
	{
		private readonly BoundExpression _Expression;

		public BoundExpression Expression => _Expression;

		public BoundXmlEmbeddedExpression(SyntaxNode syntax, BoundExpression expression, TypeSymbol type, bool hasErrors = false)
			: base(BoundKind.XmlEmbeddedExpression, syntax, type, hasErrors || BoundNodeExtensions.NonNullAndHasErrors(expression))
		{
			_Expression = expression;
		}

		[DebuggerStepThrough]
		public override BoundNode Accept(BoundTreeVisitor visitor)
		{
			return visitor.VisitXmlEmbeddedExpression(this);
		}

		public BoundXmlEmbeddedExpression Update(BoundExpression expression, TypeSymbol type)
		{
			if (expression != Expression || (object)type != base.Type)
			{
				BoundXmlEmbeddedExpression boundXmlEmbeddedExpression = new BoundXmlEmbeddedExpression(base.Syntax, expression, type, base.HasErrors);
				boundXmlEmbeddedExpression.CopyAttributes(this);
				return boundXmlEmbeddedExpression;
			}
			return this;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/BoundXmlEmbeddedExpression.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
Imports System
Imports System.Diagnostics

Namespace Microsoft.CodeAnalysis.VisualBasic
	Friend NotInheritable Class BoundXmlEmbeddedExpression
		Inherits BoundExpression
		Private ReadOnly _Expression As BoundExpression

		Public ReadOnly Property Expression As BoundExpression
			Get
				Return Me._Expression
			End Get
		End Property

		Public Sub New(ByVal syntax As SyntaxNode, ByVal expression As BoundExpression, ByVal type As TypeSymbol, Optional ByVal hasErrors As Boolean = False)
			MyBase.New(BoundKind.XmlEmbeddedExpression, syntax, type, If(hasErrors, True, expression.NonNullAndHasErrors()))
			Me._Expression = expression
		End Sub

		<DebuggerStepThrough>
		Public Overrides Function Accept(ByVal visitor As BoundTreeVisitor) As BoundNode
			Return visitor.VisitXmlEmbeddedExpression(Me)
		End Function

		Public Function Update(ByVal expression As BoundExpression, ByVal type As TypeSymbol) As Microsoft.CodeAnalysis.VisualBasic.BoundXmlEmbeddedExpression
			Dim boundXmlEmbeddedExpression As Microsoft.CodeAnalysis.VisualBasic.BoundXmlEmbeddedExpression
			If (expression <> Me.Expression OrElse CObj(type) <> CObj(MyBase.Type)) Then
				Dim boundXmlEmbeddedExpression1 As Microsoft.CodeAnalysis.VisualBasic.BoundXmlEmbeddedExpression = New Microsoft.CodeAnalysis.VisualBasic.BoundXmlEmbeddedExpression(MyBase.Syntax, expression, type, MyBase.HasErrors)
				boundXmlEmbeddedExpression1.CopyAttributes(Me)
				boundXmlEmbeddedExpression = boundXmlEmbeddedExpression1
			Else
				boundXmlEmbeddedExpression = Me
			End If
			Return boundXmlEmbeddedExpression
		End Function
	End Class
End Namespace